### DISCUSSION TITLE
posts/sample_anisotropic_microfacet_brdf/

### DISCUSSION THREAD

**Discussion by giscus** [JiayinCao/AGraphicsGuyNotes#17] [Giscus-Comments]
# posts/sample_anisotropic_microfacet_brdf/

I am working on material system in my renderer recently. My old implementation of microfacet models only supports isotropic BRDF, as a result of which, it can’t render something like brushed metals in my renderer. After spending three days in my spare time to extend the system to support anisotropic microfacet BRDF, I easily noticed how much mathematics that it needs to understand all the importance sampling methods. The fact that $ \theta $ and $ \phi $ are somewhat correlated makes the importance sampling a lot more complex than isotropic model.

https://agraphicsguynotes.com/posts/sample_anisotropic_microfacet_brdf/

**Comment by infancy** [comment]
Thank you very much for your derivation
